lib/newProfile.ml

```diff
@@ -147,16 +153,27 @@ let profile name ?args f () =
     let args = Option.map (fun f -> f()) args in
     enter name ?args ();
     let mstart = Gc.quick_stat () in
+    let istart = Instr.read_counter () in
     let v = try f ()
       with e ->
         let e = Exninfo.capture e in
+        let iend = Instr.read_counter () in
         let mend = Gc.quick_stat () in
-        let args = make_mem_diff ~mstart ~mend in
+        let args =
+          Option.default [] args @
```

Comment:
Note that if we do not re-add the user-provided (optional) `args` here again, then for "command" entries there is no way to trace end event data to a particular command, and the instruction count only appears in such end events.

Reply:
Can you explain more? IIRC the UI combine the start and end `args`.

Reply:
For instance in 
![hurkens](https://github.com/coq/coq/assets/2461932/8869fd3c-704e-46bf-b28d-0928c193e7b9)
you can see both the mem stats and the `cmd`

Reply:
I am really unsure about that, hence this thread, but I thought there was nothing linking the start and end events. So, for instance, it seems to me that nothing to ties the instruction count for a "command" (appearing in its end bit) to a specific command in the file (since information is in the "args" of the start bit).

How can the tool know what end event corresponds to what start event? Is is just a well-bracketing thing? (Sound like that would not be enough, especially since the format can also deal with concurrency.)

Reply:
It is a well bracketing thing. Concurrency is handled by using the thread id field.

Reply:
https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/edit#heading=h.yr4qxyxotyw
>If you provide args to both the B and E events then the arguments will be merged.

Reply:
Cool, thanks!